src/api/app/services/workflows/filter.rb

```diff
@@ -2,12 +2,12 @@ module Workflows
   class Filter
     def initialize(filters:)
       filters ||= {}
-      @repository_filters = filters[:repositories]
-      @architecture_filters = filters[:architectures]
+      @repository_filters = filters[:repositories].deep_symbolize_keys
+      @architecture_filters = filters[:architectures].deep_symbolize_keys
     end
 
     def match?(event)
-      match_repository?(event.payload['repository']) && match_architecture?(event.payload['arch'])
+      match_repository?(event.payload.deep_symbolize_keys[:repository]) && match_architecture?(event.payload.deep_symbolize_keys[:arch])
```

Comment:
Why not do `deep_symbolize_keys` once and store the result?

Reply:
As discussed, I'm staying with strings where values are coming from DB.